src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1164,6 +1177,57 @@ def _create_client_secret():
     return client_secret
 
 
+def _ensure_aks_acr(cli_ctx,
+                    acr_name,
+                    location,
+                    client_id,
+                    resource_group_name,
+                    subscription_id):
+    acr_client = cf_container_registry_service(cli_ctx)
+    # Generate a default ACR name.
+    registry_name = acr_name
+    if registry_name is None or registry_name == "":
+        alnum_name = ''.join(s for s in resource_group_name if s.isalnum())
+        registry_name = "aks" + alnum_name + "acr"
```

Comment:
even with resourceName, it's probably still not unique. It will report errors when the name has already taken by others, so that customer can choose another one by --acr-name.

Reply:
synced offline, this is by design.